File: leetcode/242.valid-anagram/go/isanagram_test.go
package validanagram

import "testing"

type testcase struct {
	s    string
	t    string
	want bool
}

func TestIsAnagram(t *testing.T) {
	tcs := []testcase{
		{s: "anagram", t: "nagaram", want: true},
		{s: "rat", t: "car", want: false},
		{s: "ra", t: "ard", want: false},
	}

	for _, tc := range tcs {
		t.Run("isanagram", func(t *testing.T) {
			got := isAnagram(tc.s, tc.t)
			if got != tc.want {
				t.Fatalf("isanagram, want: %v, got: %v", tc.want, got)
			}
		})
	}
}

File: leetcode/242.valid-anagram/ts/isanagram.test.ts
import { test, expect } from "vitest"
import isAnagram from "./isanagram.js"

interface testcase {
    t: string
    s: string
    want: boolean
}

test("isanagram", () => {
    const tcs: testcase[] = [
        { t: "anagram", s: "nagaram", want: true },
        { t: "rat", s: "cat", want: false },
        { t: "ra", s: "ard", want: false },
    ]

    for (const tc of tcs) {
        const got = isAnagram(tc.s, tc.t)
        expect(tc.want).toEqual(got)
    }
})
